### PR TITLE
Bugfix refactoring and changed types

### PR DESCRIPTION
**Pull request by kirillvaga**
changed types

### CODE REVIEW

File: my-app/src/API/api.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export const URL = 'https://newsapi.org/v2/top-headlines?sources=techcrunch&apiKey=251501d9720d4d64aa038e7977116df5';
```
Comment (AliaksandrZahorski) [inline my-app/src/API/api.ts]: EOL

File: my-app/src/actions/newsAction.ts [hunk 2]
```diff
@@ -1,25 +1,31 @@
...
+import { Action } from 'redux'
+import { URL } from '../API/api'
+import { FETCH_NEWS_START, FETCH_NEWS_SUCCESS, FETCH_NEWS_FAIL } from '../const/const'
 
-export const FETCH_NEWS_START = 'FETCH_NEWS_START';
-export const FETCH_NEWS_SUCCESS = 'FETCH_NEWS_SUCCESS';
-export const FETCH_NEWS_FAIL = 'FETCH_NEWS_FAIL';
-export const URL = 'https://newsapi.org/v2/top-headlines?sources=techcrunch&apiKey=251501d9720d4d64aa038e7977116df5';
+type responseType = {
+  status: string
```
Comment (AliaksandrZahorski) [inline my-app/src/actions/newsAction.ts]: semicolon, please add Prettier or similar to formatting

File: my-app/src/actions/newsAction.ts [hunk 3]
```diff
@@ -1,25 +1,31 @@
...
-        type: FETCH_NEWS_FAIL,
-        payload: { error }
-      })
-  );
+      }
+    ),
+      error =>
+        dispatch({
+          type: FETCH_NEWS_FAIL,
+          payload: { error }
```
Comment (AliaksandrZahorski) [inline my-app/src/actions/newsAction.ts]: error type?

File: my-app/src/components/App/App.tsx [hunk 4]
```diff
@@ -1,21 +1,15 @@
 import React from 'react';
-import Header from '../Header/Header'
+import { Header } from '../Header/Header'
 import Main from '../Main/Main'
-import FooterSection from '../Footer/Footer'
+import { FooterSection } from '../Footer/Footer'
 
+const myState = {
```
Comment (AliaksandrZahorski) [inline my-app/src/components/App/App.tsx]: bad naming

File: my-app/src/components/App/App.tsx [hunk 5]
```diff
@@ -1,21 +1,15 @@
...
+  userName: {
+    firstName: 'Kirill',
+    lastName: 'Vaha'
+  }
+};
 
-type AppProps = {
-  userInfo: any;
+export const App = () => {
+  return (<><Header userInfo={myState.userName} /><Main /><FooterSection /></>);
```
Comment (AliaksandrZahorski) [inline my-app/src/components/App/App.tsx]: redundant parentheses

File: my-app/src/components/CardVue/CardVue.style.ts [hunk 6]
```diff
@@ -0,0 +1,6 @@
+import styled from 'styled-components';
+
+export const WrapperForCard = styled.div`
+    background: #ECECEC;
+    padding: 30px; 
+`
```
Comment (AliaksandrZahorski) [inline my-app/src/components/CardVue/CardVue.style.ts]: EOL

File: my-app/src/components/CardVue/CardVue.tsx [hunk 7]
```diff
@@ -1,24 +1,19 @@
...
 import { Card } from 'antd';
-import './index.css'
+import '../App/App.css'
+import { WrapperForCard } from './CardVue.style'
+
 
 type CardVue = {
-  key: any;
+  key: number;
   cardVue: any;
```
Comment (AliaksandrZahorski) [inline my-app/src/components/CardVue/CardVue.tsx]: any?

File: my-app/src/components/CardVue/CardVue.tsx [hunk 8]
```diff
@@ -1,24 +1,19 @@
...
-  return (
-    <div style={{ background: '#ECECEC', padding: '30px' }}>
-      <Card title={props.cardVue.title} bordered={true}>
-        <p>{props.cardVue.author}</p>
-        <p>{props.cardVue.content}</p>
-        <p>{props.cardVue.publishedAt}</p>
-      </Card>
-    </div>
-  );
+  const { title, author, content, publishedAt } = props.cardVue;
```
Comment (AliaksandrZahorski) [inline my-app/src/components/CardVue/CardVue.tsx]: can be moved to function arguments

File: my-app/src/components/Footer/Footer.style.ts [hunk 9]
```diff
@@ -0,0 +1,11 @@
+import styled from 'styled-components';
+import { Layout } from 'antd';
+const { Footer } = Layout;
+
+export const FooterWithStyles = styled(Footer) `
+textAlign: center;
```
Comment (AliaksandrZahorski) [inline my-app/src/components/Footer/Footer.style.ts]: formatting

File: my-app/src/components/Footer/Footer.tsx [hunk 10]
```diff
@@ -1,13 +1,6 @@
...
-            Ant Design ©2016 Created by Ant UED
-        </Footer>
-    );
-}
-
-export default FooterSection;
\ No newline at end of file
+export const FooterSection = () => {
+    return (<FooterWithStyles>Ant Design ©2016 Created by Ant UED</FooterWithStyles>);
+}
```
Comment (AliaksandrZahorski) [inline my-app/src/components/Footer/Footer.tsx]: EOL

File: my-app/src/components/Header/Header.style.ts [hunk 11]
```diff
@@ -0,0 +1,46 @@
+import styled from 'styled-components';
+
+export const SectionHeader = styled.section`
+  width: 100%;
+  height: 45px;
+  background-color: #333333;
```
Comment (AliaksandrZahorski) [inline my-app/src/components/Header/Header.style.ts]: saw this color before, please move to common constants

File: my-app/src/components/Header/Header.style.ts [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+
+export const Link = styled.a`
+  text-decoration: none;
+  color: #979797;
+  font-family: 'Montserrat';
+`;
+
+export const LiCart = styled.li`
+  margin-left: 64px;
+  margin-right: 0px;
```
Comment (AliaksandrZahorski) [inline my-app/src/components/Header/Header.style.ts]: just margin

File: my-app/src/components/Header/Header.style.ts [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+    opacity: 0.5;
+  }
+`
+
+export const Cart = styled.div`
+  width          : 100%;
+  display        : flex;
+  align-items    : center;
+  justify-content: space-around;
+`;
```
Comment (AliaksandrZahorski) [inline my-app/src/components/Header/Header.style.ts]: EOL

File: my-app/src/components/Header/Header.tsx [hunk 14]
```diff
@@ -1,73 +1,30 @@
...
-  console.log(props.userInfo.lastName);
+export const Header = (props: HeaderProps) => {
+
+  const { firstName, lastName } = props.userInfo;
+
   return (
     <SectionHeader>
       <UlElement>
-        <LiElement><Link>{props.userInfo.firstName}{props.userInfo.lastName}</Link></LiElement>
+        <LiElement><Link>{firstName}{lastName}</Link></LiElement>
```
Comment (AliaksandrZahorski) [inline my-app/src/components/Header/Header.tsx]: formatting?

File: my-app/src/components/Header/Header.tsx [hunk 15]
```diff
@@ -1,73 +1,30 @@
...
         </LiCart>
       </UlElement>
     </SectionHeader>
-
   )
-}
-
-export default Header;
\ No newline at end of file
+}
```
Comment (AliaksandrZahorski) [inline my-app/src/components/Header/Header.tsx]: EOL

File: my-app/src/components/Main/Main.tsx [hunk 16]
```diff
@@ -1,48 +1,53 @@
...
-          cards.map((card: any, item: number) => (
-            <CardVue 
-              key = {item}
-              cardVue = {card}
-            />
-          ))}
-      </div>
-    );
-  }
+type stateType = {
```
Comment (AliaksandrZahorski) [inline my-app/src/components/Main/Main.tsx]: StateType

File: my-app/src/components/Main/Main.tsx [hunk 17]
```diff
@@ -1,48 +1,53 @@
...
+type MainProps = {
+  cards: object[],
+  isLoading: boolean,
+  getNews: Function
+}
 
-const mapDispatchToProps = (dispatch: any) => ({
-  getNews: bindActionCreators(getNews, dispatch)
-})
+function Main(props: MainProps) {
```
Comment (AliaksandrZahorski) [inline my-app/src/components/Main/Main.tsx]: arrow function

File: my-app/src/components/Main/Main.tsx [hunk 18]
```diff
@@ -1,48 +1,53 @@
...
+  const { cards, isLoading } = props;
```
Comment (AliaksandrZahorski) [inline my-app/src/components/Main/Main.tsx]: move to args

File: my-app/src/components/Main/Main.tsx [hunk 19]
```diff
@@ -1,48 +1,53 @@
...
+
+  useEffect(() => {
+    props.getNews();
```
Comment (AliaksandrZahorski) [inline my-app/src/components/Main/Main.tsx]: without props

File: my-app/src/components/Main/Main.tsx [hunk 20]
```diff
@@ -1,48 +1,53 @@
...
+  }, []);
+
+  return (
+    <div>
+      {isLoading ?
+        (<div>Loading ...</div>)
+        :
+        cards.map((card: object, item: number) => (
+          <CardVue
```
Comment (AliaksandrZahorski) [inline my-app/src/components/Main/Main.tsx:46]: key?

File: my-app/src/components/Main/Main.tsx [hunk 21]
```diff
@@ -1,48 +1,53 @@
...
+  useEffect(() => {
+    props.getNews();
+  }, []);
+
+  return (
+    <div>
+      {isLoading ?
+        (<div>Loading ...</div>)
+        :
+        cards.map((card: object, item: number) => (
```
Comment (AliaksandrZahorski) [inline my-app/src/components/Main/Main.tsx]: custom type instead of Object

File: my-app/src/components/Main/Main.tsx [hunk 22]
```diff
@@ -1,48 +1,53 @@
...
+}
 
-export default connect(mapStateToProps, mapDispatchToProps)(Main)
+export default connect(
+  (state: stateType) => ({
+    cards: state.cards.cards,
+    isLoading: state.cards.isLoading
+  }),
+  (dispatch: Function) => ({
+    getNews: () => dispatch(getNews())
```
Comment (AliaksandrZahorski) [inline my-app/src/components/Main/Main.tsx]: redux can wrap function without dispatch

File: my-app/src/const/const.ts [hunk 23]
```diff
@@ -0,0 +1,3 @@
+export const FETCH_NEWS_START = 'FETCH_NEWS_START';
+export const FETCH_NEWS_SUCCESS = 'FETCH_NEWS_SUCCESS';
+export const FETCH_NEWS_FAIL = 'FETCH_NEWS_FAIL';
```
Comment (AliaksandrZahorski) [inline my-app/src/const/const.ts]: EOL

File: my-app/src/index.tsx [hunk 24]
```diff
@@ -1,23 +1,11 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import App from './components/App/App';
+import {App} from './components/App/App';
```
Comment (AliaksandrZahorski) [inline my-app/src/index.tsx]: formatting

File: my-app/src/reducers/news.ts [hunk 25]
```diff
@@ -1,24 +1,30 @@
-import {FETCH_NEWS_START, FETCH_NEWS_SUCCESS, FETCH_NEWS_FAIL} from '../actions/newsAction'
+import { FETCH_NEWS_START, FETCH_NEWS_SUCCESS, FETCH_NEWS_FAIL } from '../const/const'
```
Comment (AliaksandrZahorski) [inline my-app/src/reducers/news.ts]: why two const ?

File: my-app/src/reducers/news.ts [hunk 26]
```diff
@@ -1,24 +1,30 @@
...
+
+type actionType = {
+  type: string
+  payload: Object
```
Comment (AliaksandrZahorski) [inline my-app/src/reducers/news.ts]: custom type?

**Review by AliaksandrZahorski — CHANGES_REQUESTED** [review]

File: my-app/src/reducers/news.ts (hunk 25 above)
Comment (kirillvaga) [reply my-app/src/reducers/news.ts]: first constant is the folder, second is the file

Comment (AliaksandrZahorski) [reply my-app/src/reducers/news.ts]: apiConstants ?

**Comment by Hellevar** [comment]
Is it worth to use some popular code formatting rules/tools? Like ts(es)lint

**Pull request merged**